### PR TITLE
fuzz: Minor improvements to tx_package_eval target

### PR DESCRIPTION
**Pull request by instagibbs**
Exercises `DIFFERENT_WITNESS` by using "blank" WSH() and allowing witness to determine wtxid, and attempts to make invalid/duplicate inputs.

### CODE REVIEW

File: src/test/fuzz/package_eval.cpp [hunk 1]
```diff
@@ -34,13 +34,26 @@ struct MockedTxPool : public CTxMemPool {
...
+static const CScript P2WSH_EMPTY{
+    CScript{}
+    << OP_0
+    << ToByteVector([] {
+           uint256 hash;
+           CSHA256().Write(EMPTY.data(), EMPTY.size()).Finalize(hash.begin());
+           return hash;
+       }())};
+static const std::vector<std::vector<uint8_t>> P2WSH_EMPTY_TRUE_STACK{{static_cast<uint8_t>(OP_TRUE)}, {}};
+static const std::vector<std::vector<uint8_t>> P2WSH_EMPTY_TWO_STACK{{static_cast<uint8_t>(OP_2)}, {}};
```
[the remark this answers is not in the document]
Comment (instagibbs) [reply src/test/fuzz/package_eval.cpp]: done

File: src/test/fuzz/package_eval.cpp [hunk 2]
```diff
@@ -205,6 +205,18 @@ FUZZ_TARGET(tx_package_eval, .init = initialize_tx_pool)
...
+
+                // Duplicate an input
+                bool dup_input = fuzzed_data_provider.ConsumeBool();
+                if (dup_input) {
+                    tx_mut.vin.push_back(tx_mut.vin.back());
+                }
+
+                // Refer to a non-existant input
+                if (fuzzed_data_provider.ConsumeBool()) {
+                    tx_mut.vin.emplace_back();
```
Comment (glozow) [inline src/test/fuzz/package_eval.cpp:217]: is this supposed to be empty?

Comment (instagibbs) [reply src/test/fuzz/package_eval.cpp:217]: yes, with the idea that the fuzzer won't be able to generate the preimage to all genuinely create null uint256/prevout

**Review by dergoegge — APPROVED** [review]
https://dergoegge.github.io/bitcoin-coverage/pr28825/fuzz.coverage/src/validation.cpp.gcov.html

Coverage looks good to me ACK 6a917918b76eef154c6757fe9ecf7713d526c3dd

**Review by maflcko — APPROVED** [review]
Nice

File: src/test/util/script.h [hunk 3]
```diff
@@ -18,6 +18,18 @@ static const CScript P2WSH_OP_TRUE{
...
+static const CScript P2WSH_EMPTY{
+    CScript{}
+    << OP_0
+    << ToByteVector([] {
+           uint256 hash;
+           CSHA256().Write(EMPTY.data(), EMPTY.size()).Finalize(hash.begin());
+           return hash;
+       }())};
+static const std::vector<std::vector<uint8_t>> P2WSH_EMPTY_TRUE_STACK{{static_cast<uint8_t>(OP_TRUE)}, {}};
+static const std::vector<std::vector<uint8_t>> P2WSH_EMPTY_TWO_STACK{{static_cast<uint8_t>(OP_2)}, {}};
```
[the remark this answers is not in the document]
Comment (instagibbs) [reply src/test/util/script.h:31]: will do if I touch again

**Pull request merged**